packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -360,35 +376,29 @@ export default class MqttBrokerServer implements ProtocolServer {
                 resolve();
             } else {
                 // try to connect to the broker without or with credentials
-                if (this.psw === undefined) {
+                if (this.config.psw === undefined) {
```

Comment:
Shall we have an external if to check whether this.config exists? Or is is impossible?

```
if (this.config) {
 ...
}
```

Reply:
you're right `this.config` is initialized [here](https://github.com/eclipse/thingweb.node-wot/pull/656/files#diff-1b0afdf0b85e178b3f8d59ae8dba12e7ee767d9f046f20205a319a2462b16c8dR49) and it might be `undefined`. However, instead of adding the if here I prefer to change the initialization like the following: 
```js
this.config = config ?? {};
```

I think it will work better, won't it?

Reply:
fine by me also